tunneld/api.go

```diff
@@ -214,11 +245,20 @@ func (api *API) handleTunnel(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	span := trace.SpanFromContext(ctx)
-	span.SetAttributes(
-		attribute.Bool("proxy_request", true),
-		attribute.String("user", user),
-	)
+	api.pkeyCacheMu.RLock()
+	pkey, ok := api.pkeyCache[ip]
+	api.pkeyCacheMu.RUnlock()
+
+	if !ok || time.Since(pkey.lastHandshake) > api.PeerTimeout {
+		peerNotConnected(ctx, rw, r)
+		return
+	}
+
+	peer := api.wgDevice.LookupPeer(pkey.key)
+	if peer == nil {
+		peerNotConnected(ctx, rw, r)
+		return
+	}
```

Comment:
This step seems unnecessary since we're already checking the map right?

Reply:
yeah